packages/@aws-cdk/aws-ecs/lib/firlens-log-router.ts

```diff
@@ -0,0 +1,204 @@
+   */
+  readonly firelensConfig: FirelensConfig;
+}
+
+/**
+ * Render to CfnTaskDefinition.FirelensConfigurationProperty from FirelensConfig
+ */
+function renderFirelensConfig(firelensConfig: FirelensConfig): CfnTaskDefinition.FirelensConfigurationProperty {
+  if (!firelensConfig.options) {
+    return { type: firelensConfig.type };
+  } else {
+    const enableECSLogMetadata = firelensConfig.options.enableECSLogMetadata || firelensConfig.options.enableECSLogMetadata === undefined ? 'true' : 'false';
+    const configFileType = firelensConfig.options.configFileType || /arn:aws[a-zA-Z-]*:s3:::.+/.test(firelensConfig.options.configFileValue)
+      ? FirelensConfigFileType.S3 : FirelensConfigFileType.FILE;
+    return {
+      type: firelensConfig.type,
+      options: {
+        'enable-ecs-log-metadata': enableECSLogMetadata,
+        'config-file-type': configFileType,
+        'config-file-value': firelensConfig.options.configFileValue
+      }
+    };
+  }
+}
+
+/**
+ * SSM parameters for latest fluent bit docker image in ECR
+ * https://github.com/aws/aws-for-fluent-bit#using-ssm-to-find-available-versions
+ */
+const latestFluentBitImage = '/aws/service/aws-for-fluent-bit/2.1.0';
+
+/**
+ * Obtain Fluent Bit image in Amazon ECR build and setup corresponding IAM permissions.
+ * ECR image pull permissions will be granted in task execution role.
+ * Cloudwatch logs or Firehose permissions will be grant by check options in logDriverConfig.
+ * https://docs.aws.amazon.com/AmazonECS/latest/developerguide/using_firelens.html#firelens-using-fluentbit
+ */
+export function obtainDefaultFluentBitECRImage(task: TaskDefinition, logDriverConfig?: LogDriverConfig): ContainerImage {
+  // grant ECR image pull permissions to executor role
+  task.addToExecutionRolePolicy(new iam.PolicyStatement({
+    actions: [
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage"
+    ],
+    resources: ['*']
+  }));
+
+  // grant cloudwatch or firehose permissions to task role
+  const logName = logDriverConfig && logDriverConfig.logDriver === 'awsfirelens'
+    && logDriverConfig.options && logDriverConfig.options.Name;
+  if (logName === 'cloudwatch') {
+    task.addToTaskRolePolicy(new iam.PolicyStatement({
+      actions: [
+        'logs:CreateLogGroup',
+        'logs:CreateLogStream',
+        'logs:DescribeLogStreams',
+        'logs:PutLogEvents'
+      ],
+      resources: ['*']
+    }));
+  } else if (logName === 'firehose') {
+    task.addToTaskRolePolicy(new iam.PolicyStatement({
+      actions: [
+        'firehose:PutRecordBatch',
+      ],
+      resources: ['*']
+    }));
+  }
+
+  // Not use ContainerImage.fromEcrRepository since it's not support parsing ECR repo URI,
+  // use repo ARN might result in complex Fn:: functions in cloudformation template.
+  return ContainerImage.fromRegistry(ssm.StringParameter.valueForStringParameter(task, latestFluentBitImage));
```

Comment:
Does `ssm.StringParameter.valueForStringParameter` pull the parameter at build time? (when synth is run and the CFN is generated) 

Because then you could do something smart here. Hardcoding a version is bad, and pinning to the latest tag is also bad. It might be better to instead query SSM to find the latest version at synth time, and then pin to that version. When you run synth, if the latest version is "2.1.0", the template locks to that version. That'd require some more complicated code though. This is just an idea to consider.

Reply:
If you can not do the above (or do not want to), I am personally okay with pinning to the latest tag.

Reply:
added image tag parameter in this function